File: backend/app/api/email.py
"""
文件名: app/api/email.py
创建时间: 2026-06-26
作者: TalentSense Team
功能描述: 邮件路由（模板 CRUD + 发送 + SMTP 配置），对应 API-Design.md 六
入参: HTTP 请求（to_email / candidate_ids / config / template）
出参: 统一响应 success/error
"""
from fastapi import APIRouter, Depends
from app.services.email_service import EmailService
from app.services.email_template_service import EmailTemplateService
from app.api.deps import get_current_user, require_admin
from app.models.email import (
    TemplateCreateRequest, TemplateUpdateRequest,
    SendMailRequest, SendTestRequest,
)
from app.core.response import success

router = APIRouter()


@router.post("/send")
async def send_mail(body: SendMailRequest, user: dict = Depends(get_current_user)):
    """发送邮件（模板或自定义）"""
    result = await EmailService().send_mail(
        to_email=body.to_email, template_id=body.template_id,
        custom_subject=body.custom_subject, custom_body=body.custom_body,
        variables=body.variables,
    )
    return success(data=result)


@router.post("/send-test")
async def send_test(body: SendTestRequest, user: dict = Depends(require_admin)):
    """发送测试邮件（admin only）"""
    result = await EmailService().send_test(to_email=body.to_email)
    return success(data=result)


@router.get("/templates")
async def list_templates(category: str | None = None, user: dict = Depends(get_current_user)):
    """邮件模板列表（登录用户可查，admin 才能增删改）"""
    result = await EmailTemplateService().list_templates(category=category)
    return success(data=result)


@router.post("/templates")
async def create_template(body: TemplateCreateRequest, user: dict = Depends(require_admin)):
    """创建模板（admin only）"""
    result = await EmailTemplateService().create_template(
        name=body.name, subject=body.subject, body=body.body, category=body.category,
    )
    return success(data=result)


@router.put("/templates/{template_id}")
async def update_template(template_id: str, body: TemplateUpdateRequest, user: dict = Depends(require_admin)):
    """更新模板（admin only，预置模板不可改）"""
    result = await EmailTemplateService().update_template(
        template_id, name=body.name, subject=body.subject, body=body.body, category=body.category,
    )
    return success(data=result)


@router.delete("/templates/{template_id}")
async def delete_template(template_id: str, user: dict = Depends(require_admin)):
    """删除模板（admin only，预置模板不可删）"""
    await EmailTemplateService().delete_template(template_id)
    return success()


@router.get("/config")
async def get_config(user: dict = Depends(require_admin)):
    """AC18.1: 获取 SMTP 配置（脱敏，仅 admin）"""
    result = await EmailService().get_config()
    return success(data=result)


@router.put("/config")
async def update_config(body: dict, user: dict = Depends(require_admin)):
    """AC18.2: 更新 SMTP 配置（加密存储，仅 admin）"""
    await EmailService().update_config(body)
    return success()

File: frontend/src/api/email.ts
/**
 * 文件名: api/email.ts
 * 创建时间: 2026-06-26
 * 作者: TalentSense Team
 * 功能描述: 邮件相关接口（发送 / 测试 / SMTP 配置 / 模板 CRUD）
 */
import request from './request'
import type {
  EmailConfig,
  SendMailPayload,
  SendMailResult,
  SendTestPayload,
  TemplateCreatePayload,
  TemplateItem,
  TemplateUpdatePayload,
} from '@/types/email'

/**
 * 获取邮件配置（脱敏，admin only）
 */
export function getConfig(): Promise<EmailConfig> {
  return request.get('/email/config') as unknown as Promise<EmailConfig>
}

/**
 * 更新邮件配置（admin only）
 * @param data 配置项（部分字段）
 */
export function updateConfig(data: Partial<EmailConfig>): Promise<EmailConfig> {
  return request.put('/email/config', data) as unknown as Promise<EmailConfig>
}

/**
 * 发送邮件（模板或自定义）
 * @param data SendMailPayload（template_id 与 custom_* 二选一）
 */
export function sendMail(data: SendMailPayload): Promise<SendMailResult> {
  return request.post('/email/send', data) as unknown as Promise<SendMailResult>
}

/**
 * 发送测试邮件（admin only）
 * @param data SendTestPayload（仅 to_email）
 */
export function sendTestMail(data: SendTestPayload): Promise<SendMailResult> {
  return request.post('/email/send-test', data) as unknown as Promise<SendMailResult>
}

/**
 * 模板列表（登录用户可查，admin 才能增删改）
 * @param category 可选分类筛选
 * @returns 模板数组（后端返回 {list, total}，此处剥离出 list）
 */
export async function listTemplates(category?: string): Promise<TemplateItem[]> {
  const res = await request.get('/email/templates', { params: category ? { category } : {} }) as unknown as { list: TemplateItem[]; total: number }
  return res?.list ?? []
}

/**
 * 创建模板（admin only）
 * @param data TemplateCreatePayload
 */
export function createTemplate(data: TemplateCreatePayload): Promise<TemplateItem> {
  return request.post('/email/templates', data) as unknown as Promise<TemplateItem>
}

/**
 * 更新模板（admin only，预置模板不可改）
 * @param templateId 模板 ID
 * @param data TemplateUpdatePayload（部分字段）
 */
export function updateTemplate(templateId: string, data: TemplateUpdatePayload): Promise<TemplateItem> {
  return request.put(`/email/templates/${templateId}`, data) as unknown as Promise<TemplateItem>
}

/**
 * 删除模板（admin only，预置模板不可删）
 * @param templateId 模板 ID
 */
export function deleteTemplate(templateId: string): Promise<void> {
  return request.delete(`/email/templates/${templateId}`) as unknown as Promise<void>
}
